src/Wallet/WalletManagerImpl.cpp

```diff
@@ -141,13 +147,43 @@ std::optional<TorAddress> WalletManager::AddTorListener(const SessionToken& toke
 	KeyChain keyChain = KeyChain::FromSeed(m_sessionManager.Read()->GetSeed(token));
 	ed25519_keypair_t torKey = keyChain.DeriveED25519Key(path);
 
-	TorAddress tor_address = pTorProcess->AddListener(torKey.secret_key, wallet.Read()->GetListenerPort());
-	wallet.Write()->SetTorAddress(tor_address);
-	walletImpl.Write()->SetTorAddress(tor_address);
+	TorAddress torAddress = TorAddressParser::FromPubKey(ED25519::CalculatePubKey(torKey.secret_key));
+	
+	wallet.Write()->SetTorAddress(torAddress);
+	walletImpl.Write()->SetTorAddress(torAddress);
+
+	pTorProcess->AddListener(torKey.secret_key, wallet.Read()->GetListenerPort());
 
 	return wallet.Read()->GetTorAddress();
 }
 
+bool WalletManager::RemoveCurrentTorListener(const SessionToken& token, const TorProcess::Ptr& pTorProcess)
+{
+	Locked<Wallet> wallet = m_sessionManager.Read()->GetWallet(token);
+	Locked<WalletImpl> walletImpl = m_sessionManager.Read()->GetWalletImpl(token);
+
+	KeyChain keyChain = KeyChain::FromSeed(m_sessionManager.Read()->GetSeed(token));
+	
+	int currentIndex = wallet.Read()->GetDatabase().Read()->GetCurrentAddressIndex(KeyChainPath::FromString("m/0/0"));
+	KeyChainPath currentPath = KeyChainPath::FromString("m/0/1").GetChild(currentIndex);
+	ed25519_keypair_t torKey = keyChain.DeriveED25519Key(currentPath);
+	
+	return pTorProcess->RemoveListener(TorAddressParser::FromPubKey(ED25519::CalculatePubKey(torKey.secret_key)));
+}
+
+KeyChainPath WalletManager::IncreaseKeyChainPathIndex(const SessionToken& token)
+{
+	Locked<Wallet> wallet = m_sessionManager.Read()->GetWallet(token);
+	Locked<WalletImpl> walletImpl = m_sessionManager.Read()->GetWalletImpl(token);
+
+	KeyChainPath userPath = KeyChainPath::FromString("m/0/0"); // FUTURE: Support multiple account paths
+	
+	// now let's increase the index
+	int nextIndex = wallet.Write()->GetDatabase().Write()->IncreaseAddressIndex(userPath);
```

Comment:
You're just trying to increase address index for the tor listener, not for the next blinding factor generation. So just pass `KeyChainPath::FromString("m/0/1")` into `IncreaseAddressIndex` instead.